### PR TITLE
Fix boundary filtering to Any?

### PR DESCRIPTION
**Pull request by hzsweers**

### CODE REVIEW

File: src/main/java/com/squareup/kotlinpoet/TypeVariableName.kt [hunk 1]
```diff
@@ -56,12 +56,14 @@ class TypeVariableName private constructor(
   override fun emit(out: CodeWriter) = out.emit(name)
 
   companion object {
+    private val NULLABLE_ANY = ANY.asNullable()
```
Comment (Egorand) [inline src/main/java/com/squareup/kotlinpoet/TypeVariableName.kt]: Should we add that to the other `ClassName`s we define in `TypeName`? Also, `ANY?` is an option for the field name, `NULLABLE_ANY` is totally fine though.

Comment (hzsweers) [reply src/main/java/com/squareup/kotlinpoet/TypeVariableName.kt]: Up to y'all!

**Comment by hzsweers** [comment]
I can add a test

File: src/main/java/com/squareup/kotlinpoet/TypeVariableName.kt (hunk 1 above)
Comment (Egorand) [reply src/main/java/com/squareup/kotlinpoet/TypeVariableName.kt]: Sorry, wanted to say `` `ANY?` ``

Comment (hzsweers) [reply src/main/java/com/squareup/kotlinpoet/TypeVariableName.kt]: I'll leave this along for now and allow y'all to do a follow up if people want

**Pull request merged**